Cinebase/Use Cases/Movies/MovieDetail/TagSelectionControl.swift

```diff
@@ -0,0 +1,375 @@
+    
+    open var selectedTextColor: UIColor? {
+        didSet {
+            reloadStyles()
+        }
+    }
+    
+    open var tagBackgroundColor: UIColor = UIColor.gray {
+        didSet {
+            reloadStyles()
+        }
+    }
+    
+    open var selectedBackgroundColor: UIColor? {
+        didSet {
+            reloadStyles()
+        }
+    }
+    
+    var textFont: UIFont = UIFont.systemFont(ofSize: 12) {
+        didSet {
+            tagNameLabel.font = textFont
+        }
+    }
+    
+    var paddingY: CGFloat = 2
+    var paddingX: CGFloat = 2
+    
+    private func reloadStyles() {
+        if isSelected {
+            backgroundColor = selectedBackgroundColor ?? tagBackgroundColor
+            tagNameLabel.textColor = selectedTextColor ?? textColor
+        }
+        else {
+            backgroundColor = tagBackgroundColor
+            tagNameLabel.textColor = textColor
+        }
+    }
+    
+    weak var delegate: TagViewDelegate?
+    
+    var tagHeight: CGFloat?
+    
+    var isSquare: Bool = false
+    
+    init(tagName: String, height: CGFloat? = nil, square: Bool = false) {
+        
+        if let height = height {
+           self.tagHeight = height
+        }
+        self.isSquare = square
+        super.init(frame: CGRect.zero)
+        tagNameLabel.text = tagName
+        
+        self.addTarget(self, action: #selector(viewTapped), for: .touchDown)
+        reloadStyles()
+    }
+
+    required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    func viewTapped(_ sender: UIControl) {
+        delegate?.tagViewTapped(view: self)
+    }
+    
+    override open var intrinsicContentSize: CGSize {
+        
+        var size = tagNameLabel.text?.size(attributes: [NSFontAttributeName: textFont]) ?? CGSize.zero
+        size.width += paddingX * 2 + cornerRadius * 2
+        
+        if let height = tagHeight {
+            return isSquare ? CGSize(width: height, height: height) : CGSize(width: size.width, height: height)
+        } else {
+            size.height = textFont.pointSize + paddingY * 2
+            return isSquare ? CGSize(width: size.width, height: size.width) : CGSize(width: size.width < size.height ? size.height : size.width, height: size.height)
+        }
+        
+    }
+
+    override func layoutSubviews() {
+        
+        super.layoutSubviews()
+        
+        if isSquare {
+            let width = self.bounds.width - paddingX * 2 - cornerRadius * 2
+            tagNameLabel.frame = CGRect(x: cornerRadius + paddingX, y: paddingY, width: width, height: self.bounds.height - paddingY * 2)
+            
+        } else {
+            let size = tagNameLabel.text?.size(attributes: [NSFontAttributeName: textFont]) ?? CGSize.zero
+            tagNameLabel.frame = CGRect(x: cornerRadius + paddingX, y: self.bounds.midY - size.height / 2, width: size.width, height: size.height)
+        }
+        addSubview(tagNameLabel)
```

Comment:
I believe we only need to add the label as a subview once. We can move this line to the `init`